### PR TITLE
Maat.model

### PR DESCRIPTION
**Pull request by serbinsh**
More updates to code.  Enabled runs with met drivers.  Tested at Fort Dix and BCI, both with AF and custom met drivers from BCI tower

Working on a util to convert from ed2 to maat drivers, but may not follow through on that...

### CODE REVIEW

**Comment by mdietze** [comment]
@serbinsh why would you want to convert from ED2 met to MAAT met, rather than converting from CF to MAAT like normal? It's not like there's some huge log of ED2 met drivers for sites we don't have CF (I'm unaware of a single site in that category)

File: models/maat/R/model2netcdf.MAAT.R [hunk 1]
```diff
@@ -37,72 +45,66 @@
...
+    output[[1]] <- out.year                       # Simulation year
+    output[[2]] <- sub.maat.doy+day.steps         # Fractional day - NEED TO IMPLEMENT  
+    output[[3]] <- (sub.maat.output$A)            # assimilation in umolsC/m2/s
+    
+    ## TODO: ADD MORE MAAT OUTPUTS HERE ##
     
     #******************** Declare netCDF variables ********************#
-    ## TODO !!!THIS BIT NEEDS UPDATING TO CAPTURE HIGH-FREQUENCY (sub daily) OUTPUTS !!!
+    ## This version doesn't provide enough output timesteps when running with met data that has
+    ## a step greater than 1 per day
```
Comment (mdietze) [inline models/maat/R/model2netcdf.MAAT.R]: You might want to throw an error or warning explicitly if you detect a timestep you can't process.  Hope to have all temporal downscaling handled BEFORE met2model functions using common tools, so you shouldn't ever have to implement that downscaling internally.

**Comment by serbinsh** [comment]
@mcdietze  Its a long story but before I had access to the raw BCI data so i could create a met driver I was going to write some code to convert ED2 to MAAT.  I needed a test driver.  I then started playing with Ameriflux instead to get it running but because I still have the FTP firewall issue I can’t get data for sites outside of the US using met process (yet).

But I think found some BCI data, created a met file, registered and am good to go.  That is why I say I would probably abandon the ed2 to met.  Which I will since I won’t really need it once I fix the FTP firewall issue.

S
------------------------------------------------------------------------------
Shawn P. Serbin
Environmental & Climate Sciences Department
Brookhaven National Laboratory
Upton, NY 11973-5000

Phone: +1 631-344-3165
Fax: +1 631-344-2060
Email: sserbin@bnl.gov<mailto:sserbin@bnl.gov>
Bio:  http://www.bnl.gov/envsci/bio/serbin-shawn.php
TEST Group: http://www.bnl.gov/TEST/
ORCiD: http://orcid.org/0000-0003-4136-8971
Twitter: @doctorjackpine, @test_bnl
------------------------------------------------------------------------------

On Sep 14, 2016, at 10:07 AM, Michael Dietze <notifications@github.com<mailto:notifications@github.com>> wrote:


@serbinsh<https://github.com/serbinsh> why would you want to convert from ED2 met to MAAT met, rather than converting from CF to MAAT like normal? It's not like there's some huge log of ED2 met drivers for sites we don't have CF (I'm unaware of a single site in that category)

**Comment by mdietze** [comment]
Does your firewall block THREEDS as well or just FTP? There's a number of met products that work without FTP.

**Comment by serbinsh** [comment]
I need to check on the THREEDS.  I suspect that will be fine....i will report back

[iPhone]

On Sep 14, 2016, at 12:30 PM, Michael Dietze <notifications@github.com<mailto:notifications@github.com>> wrote:


Does your firewall block THREEDS as well or just FTP? There's a number of met products that work without FTP.

**Pull request merged**